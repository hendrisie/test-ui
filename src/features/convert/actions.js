// @flow
import {
  ACTION_SUBMIT,
} from './constants';

export const submitTotal = (payload) => ({
  type: ACTION_SUBMIT,
  payload,
});
