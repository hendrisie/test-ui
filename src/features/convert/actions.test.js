import configureMockStore from 'redux-mock-store';
import { submitTotal } from './actions';

let mockStore;
let store;

beforeAll(() => {
  mockStore = configureMockStore();
  store = mockStore({});
});

describe('action creators', () => {
  it('creates action = ACTION_SUBMIT when submitTotal was successful', () => {
    // Dispatch the createSuccess action with the values of a new to-do.
    store.dispatch(
      submitTotal('Rp 10501')
    );
    expect(store.getActions()).toMatchSnapshot();
  });
});
