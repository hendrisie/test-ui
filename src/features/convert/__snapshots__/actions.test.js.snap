// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action creators creates action = ACTION_SUBMIT when submitTotal was successful 1`] = `
Array [
  Object {
    "payload": "Rp 10501",
    "type": "features/convert/ACTION_SUBMIT",
  },
]
`;
